apps/backend/src/groups/groups.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { Group, GroupMembers, Prisma, Role, User } from '@prisma/client';
+import { PrismaService } from 'nestjs-prisma';
+
+@Injectable()
+export class GroupsService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async create(data: Prisma.GroupCreateInput): Promise<Group> {
+    return await this.prisma.group.create({ data });
+  }
+
+  async findAll(): Promise<Group[]> {
+    if (!this.prisma.group) {
+      throw new NotFoundException('Groups not found');
+    }
+    return await this.prisma.group.findMany();
+  }
+
+  async findOne(id: number): Promise<Group> {
+    const group = await this.prisma.group.findUnique({ where: { id: id } });
+    if (!group) {
+      throw new NotFoundException(`Group with ID ${id} not found`);
+    }
+    return group;
+  }
+
+  async findMembers(id: number): Promise<User[]> {
+    const group = await this.prisma.group.findUnique({ where: { id }, include: { members: true } });
+    if (!group) {
+      throw new NotFoundException(`Group with ID ${id} not found`);
+    }
+    return this.prisma.user.findMany({ where: { id: { in: group.members.map((member) => member.userId) } } });
+  }
+
+  async update(id: number, data: Prisma.GroupUpdateInput): Promise<Group> {
+    try {
+      return await this.prisma.group.update({ where: { id }, data });
+    } catch {
+      throw new NotFoundException(`Group with ID ${id} could not be updated`);
+    }
+  }
+
+  async remove(id: number) {
+    try {
+      return await this.prisma.group.delete({ where: { id } });
+    } catch {
+      throw new NotFoundException(`Group with ID ${id} could not be deleted`);
+    }
+  }
+
+  async addMember(groupId: number, userId: number): Promise<GroupMembers> {
+    const groupMember = await this.prisma.groupMembers.create({
```

Comment:
Try-catch would be useful here, too, like when the user is already added to the group

Reply:
and that's true for most of the functions (update, remove, ..)